packages/wonder-blocks-banner/src/components/banner.js

```diff
@@ -0,0 +1,281 @@
+// @flow
+import * as React from "react";
+import {StyleSheet, css} from "aphrodite";
+
+import Button from "@khanacademy/wonder-blocks-button";
+import Color from "@khanacademy/wonder-blocks-color";
+import {View} from "@khanacademy/wonder-blocks-core";
+import {icons} from "@khanacademy/wonder-blocks-icon";
+import Link from "@khanacademy/wonder-blocks-link";
+import IconButton from "@khanacademy/wonder-blocks-icon-button";
+import Spacing from "@khanacademy/wonder-blocks-spacing";
+import {LabelSmall} from "@khanacademy/wonder-blocks-typography";
+
+type BannerKind =
+    /**
+     * Color blue, circle 'i' icon. This is the default.
+     */
+    | "info"
+    /**
+     * Color green, smiley icon
+     */
+    | "success"
+    /**
+     * Color gold, triangle exclamation-point icon
+     */
+    | "warning"
+    /**
+     * Color red, circle exclamation-point icon
+     */
+    | "critical";
+
+type BannerLayout =
+    /**
+     * Renders a rounded rectangle, usually for when banner is used as an inset
+     * element on a screen (e.g., the SOT card) that appears to be floating
+     */
+    | "floating"
+    /**
+     * Renders a full-width banner, with no rounded corners.
+     * This is the default.
+     */
+    | "full-width";
+
+type ActionTrigger = {|
+    title: string,
+    onClick?: () => void,
+    href?: string,
+    ariaLabel?: string,
+|};
```

Comment:
**suggestion:** Now that I see, this type could be improved so we can enforce that an action item includes only a `onClick` or a `href` prop (not both at the same time). You can see what I refer to here:

https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVAXAngBwKZgCCAxhgJZwB2AKgE5kDmDetYAvGAN4A+qY-Ychhh4AXGADOGepQYAaPgKoBhGGWIBrAPziAFAEp2APjAA3OGQAmCgWAAWtPFB2TpZWTYEBDel4AyXgBGeDAuUjLyqNwAvgDc6MDAYNR2ZBJglnB4EpQA5Bj2XqYELLRwtBJyYIEArgVkBRJ2cDUwlmBQXmQwYGRQYAgE7sQwNZYEGKnpxF4SeKjEVFK9ALY4cBISZIEiAMoYXhhiRKQUNPRMLOxcikIi4gBEdiHwD54qapp6hmwmnNGeBxOR4SOArPAAWjUlA0b1QcQSSQAsl4NO4GKt1pttiJXIdspiNlsdvNsPgTuQqHRGMxaAAhWYEDg8W4Ne6uCKeHxkfxBEJhNweKIIzC4AgkSnnGksADqDTsdLqGCo1xZ-AAdJqJWdqZd6Yz3pRVOoNN9jGYLNZhfFReTtVSLrS5ZM-O4NKreBqtacHdL9XNAY4oOJwuiFDEbWTxT6pXqAEzXe2xp3yxUYZWUMDcCk6x2y+WumE2hZLeqUUxeNSWJPiJO62kJ5ms4THJ4vOBw-gfE1m35cAGKIHBsAPUHgqFuuEixaUZYVqtJwgSNMZ2sx+ssRs3fh3VvPGCvQ3Gr5gAzm-4KaelsyVqyLiSF00533x1XN9ltg8dwPAkdjyHQrCl6xEAA

Reply:
Oh interesting. I feel like we should provide the option to have both an `href` and an `onClick` as there are some links that run an `onClick` before navigation. For example, links that get analytics data. I guess we could just ignore that case for now and then add support for it later if someone requests it?

Reply:
I definitely like your solution way better. I just wasn't sure if there's a case that needs both href and onClick

Reply:
Here you can see an updated version that includes this valid scenario: https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoVAXAngBwKZgCCAxhgJZwB2AKgE5kDmDetYAvGAN4A+qY-Ychhh4AXGADOGepQYAaPgKoBhGGWIBrAPziAFAEp2APjAA3OGQAmCgWAAWtPFB2TpZWTYEBDel4AyXgBGeDAuUjLyqNwAvgDc6MDAYNR2ZBJglnB4EpQA5Bj2XqYELLRwtBJyYIEArgVkBRJ2cDUwlmBQXmQwYGRQYAgE7sQwNZYEGKnpxF4SeKjEVFK9ALY4cBISZIEiAMoYXhhiRKQUNPRMLOxcikIi4gBEdiHwD54qapp6hmwmnNGeBxOR4SOArPAAWjUlA0b1QcQSSQAsl4NO4GKt1pttiJXIdspiNlsdvNsPgTuQqHRGMxaAAhWYEDg8W4Ne6uCKeHxkfxBEJhNweKIIzC4AgkSnnGksADqDTsdLqGCo1xZ-AAdJqJWdqZd6Yz3pRVOoNN9jGYLNZhfFReTtVSLrS5ZM-O4NKreBqtacHdL9XNAY4oOJwujDcbNC4DObzFYFDEbWTxT6pXqAEzXe2pp3yxUYZWUMDcCk6x2y+WumE2xL2INgcr1o2fd1NFptaoEcZQdx4SwLJb1SimLxqSxZ8RZ3W0jPM1nCY5PF5wOEixaUZbD0dZwgSPMFicpqcsGc3fh3BfPGCvcPNs2-LgA+E2tcbkdWbcSSumku+9OqufsouV7LoGwJgA8oLglCbors+A5mG+Y4pjuX7OnYAASXiUJYIi0AekpHrQJ6cABF5LnC-BAsG4GQZC0KwjeJp3n80RPkAA

One nice thing of this approach is that we can get rid of the runtime validation that was defined in the component.

Reply:
That's right, and that's the purpose of static types (e.g Flow, TS)..... that we give signals that something is wrong, but that at the same time we cannot stop devs from bypassing that (if they want). If they still need to bypass that, they would need to use a `$FlowIgnore` comment, and that's something we don't recommend to do (unless there's a very specific reason to do so).